commit 2c42a787257e0801b8454fd8960d6794831cedea
Author: o9000 <mrovi9000@gmail.com>
Date:   Sat Oct 8 15:02:28 2016 +0200

    Fix compilation with ENABLE_RSVG=OFF (fixes issue #602)

diff --git a/src/launcher/launcher.c b/src/launcher/launcher.c
index 0b4172f6dc35..0052e9068beb 100644
--- a/src/launcher/launcher.c
+++ b/src/launcher/launcher.c
@@ -30,7 +30,6 @@
 #include <glib/gi18n.h>
 #include <glib.h>
 #include <glib/gstdio.h>
-#include <gdk-pixbuf/gdk-pixbuf.h>
 #include <sys/types.h>
 
 #include "window.h"
