diff --git a/server/CMakeLists.txt b/server/CMakeLists.txt
index 8429dc64c742..4517780793e5 100644
--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -29,9 +29,7 @@ if(WIN32)
 endif()
 
 # JsonCpp
-set(JSONCPP_DIR ../libraries/jsoncpp)
-file(GLOB JSONCPP_HEADERS ${JSONCPP_DIR}/json/*.h)
-set(JSONCPP_SOURCE ${JSONCPP_DIR}/jsoncpp.cpp)
+set(JSONCPP_DIR /usr/include/jsoncpp)
 
 # MiniUPnPc
 if (USE_MINIUPNP)
@@ -46,8 +44,6 @@ set_source_files_properties(${COMMON_DIR}/version.cpp PROPERTIES COMPILE_FLAGS -
 define_platform()
 
 # Server definitions
-add_definitions(-DJSON_IS_AMALGAMATION)
-
 if (USE_MINIUPNP)
   add_definitions(-DODA_HAVE_MINIUPNP)
 endif()
@@ -68,7 +64,6 @@ if(NOT APPLE AND NOT WIN32)
 endif()
 
 add_executable(odasrv
-  ${JSONCPP_SOURCE} ${JSONCPP_HEADERS}
   ${COMMON_SOURCES} ${COMMON_HEADERS}
   ${SERVER_SOURCES} ${SERVER_HEADERS}
   ${SERVER_WIN32_HEADERS} ${SERVER_WIN32_RESOURCES})
@@ -87,6 +82,8 @@ if(UNIX AND NOT APPLE)
   target_link_libraries(odasrv rt)
 endif()
 
+target_link_libraries(odasrv -ljsoncpp)
+
 if(APPLE)
 elseif(WIN32)
   install(TARGETS odasrv
